### PR TITLE
limit celerybeat to one server

### PR DESCRIPTION
**Pull request by thraxil**
In a deployment with more than one edxapp server, we need to make sure that `lms_celerybeat` and `cms_celerybeat` are only installed on one of them; otherwise we get duplicate scheduled tasks, conflicts, and general pain.

This is about the best solution I could come up with using ansible. Ansible includes a `run_once` directive, but that has the problem that it will always run it once even if one of the hosts is
temporarily unreachable during the deploy. Ie, if the one with celery beat is temporarily down during a deploy, ansible will install/configure it on one of the other hosts and later when the initial one comes back, they will both have it. The `inventory_hostname == ansible_play_hosts_all[0]` basically tells it to install it on the first host in the list as it's configured. As long as we don't change the order of entries in the `inventory`, it should be fine.

### CODE REVIEW

**Review by bryanlandia — APPROVED** [review]
Looks good @thraxil and it's good to know about the `inventory_hostname` approach.  You might consider making this PR upstream, as well.

**Comment by thraxil** [comment]
@bryanlandia upstream actually doesn't seem to ever enable celerybeat at all. The tasks that this modifies are ones that we added.

**Review by melvinsoft — COMMENTED** [review]
@thraxil This is awesome, super elegant. Thanks!

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 1]
```diff
@@ -445,15 +445,17 @@
     supervisorctl_path: "{{ supervisor_ctl }}"
     config: "{{ supervisor_cfg }}"
     state: restarted
-  when: celery_worker is defined and not disable_edx_services and EDXAPP_CELERY_BEAT_LMS_ENABLED
+  when: (celery_worker is defined and not disable_edx_services and EDXAPP_CELERY_BEAT_LMS_ENABLED) and
+        (inventory_hostname == ansible_play_hosts_all[0])
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/tasks/deploy.yml]:
```suggestion
  when:
    - celery_worker is defined and not disable_edx_services and EDXAPP_CELERY_BEAT_LMS_ENABLED
    - inventory_hostname == ansible_play_hosts_all[0]   # Avoid duplicate scheduling
```

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 2]
```diff
@@ -445,15 +445,17 @@
...
 
 - name: restart celery beat cms process
   supervisorctl:
     name: cms_celerybeat
     supervisorctl_path: "{{ supervisor_ctl }}"
     config: "{{ supervisor_cfg }}"
     state: restarted
-  when: celery_worker is defined and not disable_edx_services and EDXAPP_CELERY_BEAT_CMS_ENABLED
+  when: (celery_worker is defined and not disable_edx_services and EDXAPP_CELERY_BEAT_CMS_ENABLED) and
+        (inventory_hostname == ansible_play_hosts_all[0])
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/tasks/deploy.yml]:
```suggestion
  when:
    - celery_worker is defined and not disable_edx_services and EDXAPP_CELERY_BEAT_CMS_ENABLED
    - inventory_hostname == ansible_play_hosts_all[0]  # Avoid duplicate scheduling
```

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 3]
```diff
@@ -111,7 +111,8 @@
     - install
     - install:configuration
     - install:celerybeat
-  when: EDXAPP_CELERY_BEAT_LMS_ENABLED or EDXAPP_CELERY_BEAT_CMS_ENABLED
+  when: (EDXAPP_CELERY_BEAT_LMS_ENABLED or EDXAPP_CELERY_BEAT_CMS_ENABLED) and
+        (inventory_hostname == ansible_play_hosts_all[0])
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]:
```suggestion
  when:
   - EDXAPP_CELERY_BEAT_LMS_ENABLED or EDXAPP_CELERY_BEAT_CMS_ENABLED
   - inventory_hostname == ansible_play_hosts_all[0]  # Avoid duplicate scheduling
```

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 4]
```diff
@@ -166,7 +167,8 @@
     force: yes
   become_user: "{{ supervisor_user }}"
   when: celery_worker is not defined and not disable_edx_services and
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]:
```suggestion
```

File: playbooks/roles/edxapp/tasks/service_variant_config.yml [hunk 5]
```diff
@@ -166,7 +167,8 @@
...
-        (EDXAPP_CELERY_BEAT_LMS_ENABLED or EDXAPP_CELERY_BEAT_CMS_ENABLED)
+        (EDXAPP_CELERY_BEAT_LMS_ENABLED or EDXAPP_CELERY_BEAT_CMS_ENABLED) and
+        (inventory_hostname == ansible_play_hosts_all[0])
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/tasks/service_variant_config.yml]:
```suggestion
  when: 
    - celery_worker is not defined and not disable_edx_services
    - EDXAPP_CELERY_BEAT_LMS_ENABLED or EDXAPP_CELERY_BEAT_CMS_ENABLED
    - inventory_hostname == ansible_play_hosts_all[0]  # Avoid duplicate scheduling
```

**Review by OmarIthawi — APPROVED** [review]
Thanks Anders for finding and fixing the issue!

I've added minor suggestions to use the more verbose `when` style because I find it more readable feel free to go without it.

**Comment by bryanlandia** [comment]
@thraxil I don't think this is quite working..   see https://console.cloud.google.com/cloud-build/builds/c4a76090-e5e2-44e6-8e67-65094648c8e4?authuser=1&organizationId=508326437899&orgonly=true&project=appsembler-tahoe-0

**Comment by thraxil** [comment]
@bryanlandia yeah, I think the issue is that Tahoe staging now has the setup with all celery stuff split off onto one worker and none of it on the edxapp servers.

**Comment by thraxil** [comment]
I think @melvinsoft is deploying to staging via #290 and that conflicts with this.

**Pull request merged**